modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/causalitydatanodes/DistributionZoneCausalityDataNodesTest.java

```diff
@@ -1013,73 +997,116 @@ void simpleTopologyChanges() throws Exception {
+        expectedDataNodes1.put(getZoneId(ZONE_NAME_4), Set.of(NODE_0.name(), NODE_1.name()));
 
-        CountDownLatch scaleDownLatch = blockScaleDownTaskExecution();
+        Map<Integer, Set<String>> expectedDataNodes2 = new HashMap<>(expectedDataNodes);
+        expectedDataNodes2.put(getZoneId(ZONE_NAME_4), Set.of(NODE_0.name(), NODE_1.name(), NODE_2.name()));
 
-        Set<LogicalNode> topology1 = Set.of(NODE_0, NODE_1, NODE_2);
-        Set<LogicalNode> topology2 = Set.of(NODE_0, NODE_2);
+        checkDataNodesRepeated(expectedDataNodes1, expectedDataNodes2, true);
+    }
 
-        // Change logical topology. NODE_2 is added.
-        long topologyRevision1 = putNodeInLogicalTopologyAndGetRevision(NODE_2, topology1);
-        waitForCondition(() -> metaStorageManager.appliedRevision() >= topologyRevision1, TIMEOUT);
+    /**
+     * Check that data nodes calculation is idempotented.
+     * The current data nodes value which was calculated by data nodes from the meta storage manager and topology augmentation maps equals
+     * to the data nodes value which were calculated by data nodes from the meta storage manager after topology augmentation maps
+     * were cleared.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    void checkDataNodesRepeatedOnNodeRemoved() throws Exception {
+        prepareZonesWithTwoDataNodes();
 
-        // Change logical topology. NODE_1 is removed.
-        long topologyRevision2 = removeNodeInLogicalTopologyAndGetRevision(Set.of(NODE_1), topology2);
-        waitForCondition(() -> metaStorageManager.appliedRevision() >= topologyRevision2, TIMEOUT);
-
-        Set<String> expectedDefaultZoneDataNodes = Set.of(NODE_0.name(), NODE_2.name());
-        Set<String> expectedZone1DataNodes = Set.of(NODE_0.name(), NODE_1.name(), NODE_2.name());
-        Set<String> expectedZone2DataNodes = Set.of(NODE_0.name());
-        Set<String> expectedZone3DataNodes = Set.of(NODE_0.name(), NODE_1.name());
-
-        // Check that data nodes values are changed according to scale up and down timers.
-        // Data nodes from the meta storage manager and topology augmentation maps are used to calculate current data nodes.
-        checkDataNodes(
-                topologyRevision2,
-                expectedDefaultZoneDataNodes,
-                expectedZone1DataNodes,
-                expectedZone2DataNodes,
-                expectedZone3DataNodes
-        );
+        Map<Integer, Set<String>> expectedDataNodes = new HashMap<>();
+        expectedDataNodes.put(getZoneId(DEFAULT_ZONE_NAME), Set.of(NODE_0.name()));
+        expectedDataNodes.put(getZoneId(ZONE_NAME), Set.of(NODE_0.name(), NODE_1.name()));
+        expectedDataNodes.put(getZoneId(ZONE_NAME_2), Set.of(NODE_0.name()));
+        expectedDataNodes.put(getZoneId(ZONE_NAME_3), Set.of(NODE_0.name(), NODE_1.name()));
+
+        Map<Integer, Set<String>> expectedDataNodes1 = new HashMap<>(expectedDataNodes);
```

Comment:
Let's add meaningful names to these expected data nodes, as we did for `DistributionZoneCausalityDataNodesTest#checkDataNodesRepeated` method params

Reply:
Fixed